### PR TITLE
ci: Update dependabot actions, move dependency files to a different codeowner

### PR DESCRIPTION
**Pull request by jakemhiller**
### Overview

1. Set package.json and yarn.lock changes to the @Codecademy/dependencies group so we can control notifications for those changes
2. Update dependabot to make PRs against the main branch
3. Update dependabot scripts to enable default github automerge and disable requiring the "ship-it" label.

Workflow for managing these dependabot PRs will now be for one of us to approve the PR, it should then automerge once it passes CI.

### CODE REVIEW

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62854e8c99f64c2df3b101eb--gamut-preview.netlify.app>https://62854e8c99f64c2df3b101eb--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62854e8c99f64c2df3b101eb>Deploy Logs</a></p>

**Review by dreamwasp — APPROVED** [review]
🔥

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -2,4 +2,7 @@
 * @codecademy/gamut
 
 # -- Documentation changes
-packages/styleguide/ @songstephen @codecademy/gamut
+packages/styleguide/ @songstephen @Codecademy/gamut
+
+package.json @Codecademy/dependencies
```
Comment (jakemhiller) [inline .github/CODEOWNERS:7]: not 100% sure if i need an asterisk here to pick up the files in the packages/* folder

Comment (jakemhiller) [reply .github/CODEOWNERS:7]: i do not

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62855344e1db0d329a04cd8a--gamut-preview.netlify.app>https://62855344e1db0d329a04cd8a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62855344e1db0d329a04cd8a>Deploy Logs</a></p>

**Pull request merged**